google/cloud/bigtable/data/_async/_read_rows.py

```diff
@@ -59,158 +59,253 @@ class _ReadRowsOperationAsync(AsyncIterable[Row]):
+                            cl = c.labels
+                            k = c.row_key
+                            if (
+                                c.HasField("family_name")
+                                and c.family_name.value != family
+                            ):
+                                raise InvalidChunk("family changed mid cell")
+                            if (
+                                c.HasField("qualifier")
+                                and c.qualifier.value != qualifier
+                            ):
+                                raise InvalidChunk("qualifier changed mid cell")
+                            if t and t != ts:
+                                raise InvalidChunk("timestamp changed mid cell")
+                            if cl and cl != labels:
+                                raise InvalidChunk("labels changed mid cell")
+                            if k and k != row_key:
+                                raise InvalidChunk("row key changed mid cell")
+
+                            if c.reset_row:
+                                raise _ResetRow(c)
+                            buffer.append(c.value)
+                        value = b"".join(buffer)
+                    if family is None:
+                        raise InvalidChunk("missing family")
+                    if qualifier is None:
+                        raise InvalidChunk("missing qualifier")
+                    cells.append(
+                        Cell(value, row_key, family, qualifier, ts, list(labels))
+                    )
+                    if c.commit_row:
+                        yield Row(row_key, cells)
+                        break
+                    c = await it.__anext__()
+            except _ResetRow as e:
+                c = e.chunk
                 if (
-                    self._last_emitted_row_key is not None
-                    and new_item.row_key <= self._last_emitted_row_key
+                    c.row_key
+                    or c.HasField("family_name")
+                    or c.HasField("qualifier")
+                    or c.timestamp_micros
+                    or c.labels
+                    or c.value
                 ):
-                    raise InvalidChunk("Last emitted row key out of order")
-                # don't yeild _LastScannedRow markers; they
-                # should only update last_seen_row_key
-                if not isinstance(new_item, _LastScannedRow):
-                    yield new_item
-                    self._emit_count += 1
-                self._last_emitted_row_key = new_item.row_key
-                if total_row_limit and self._emit_count >= total_row_limit:
-                    return
-        except (Exception, GeneratorExit) as exc:
-            # ensure grpc stream is closed
-            new_gapic_stream.cancel()
-            raise exc
+                    raise InvalidChunk("reset row with data")
+                continue
```

Comment:
I think you have to reset the shared family & qualifier variables

Reply:
Those should be reset on each row iteration on L232 and L233